### ISSUE TITLE
Incorrect processing of plot_args when using data argument

### ISSUE THREAD

**Issue by timhoffm** [matplotlib/matplotlib#11389]
### Bug report

As opposed to dicts, pandas.DataFramse etc. `name in container` does not check if a field exists in a numpy array. Hence, mustn't do `args[1] in data` unconditionally on the data type. This was spotted due to the indirectly related warning:

~~~
lib/matplotlib/tests/test_axes.py::test_structured_data
  /home/travis/build/matplotlib/matplotlib/lib/matplotlib/axes/_axes.py:52: FutureWarning: elementwise == comparison failed and returning scalar instead; this will raise an error or perform elementwise comparison in the future.
    if (not args[1] in data and
~~~

Possible fixes:
- #10872, which gets the logic straight
- or alternatively #10928, which is a larger cleanup and removes `_plot_args_replacer()` altogether.

**Comment by jklymak** [comment]
@timhoffm is this closed by #10872?

**Comment by timhoffm** [comment]
Yes, fixed by #10872.
